src/use_case/toggle_audio/ToggleAudioInteractor.java

```diff
@@ -27,14 +28,53 @@ public void execute(ToggleAudioInputData inputData) {
         boolean audioPlaying = songAudio.isPlaying();
 
         if (audioPlaying) {
+            cancelTimerForShowingPlayButton();
+
             //stopping audio and showing play button
             songAudio.stop();
             toggleAudioPresenter.showPlayButton();
         }
         else {
-            //playing audio and showing pause button
+            startTimerForShowingPlayButton(songAudio);
+
+            // playing audio and showing pause button
             songAudio.play();
             toggleAudioPresenter.showPauseButton();
         }
     }
+
+    private void startTimerForShowingPlayButton(PlayableAudio audio) {
+        cancelTimerForShowingPlayButton();
+
+        checkToggleStateThread = new Thread(() -> {
+            try {
+                int timeBetweenChecks = 20;
+                int audioMaxDuration = 10000;
+                int buffer = 500;
+
+                int i = 0;
+                while (i < audioMaxDuration + buffer) {
+                    Thread.sleep(timeBetweenChecks);
+                    if (Thread.interrupted()) {
+                        return;
+                    }
+                    i += timeBetweenChecks;
+                }
+
+                if (!audio.isPlaying()) {
+                    toggleAudioPresenter.showPlayButton();
+                }
+            } catch (InterruptedException ignored) {
+            }
+        });
+        checkToggleStateThread.start();
+    }
```

Comment:
Could you explain how fixing Known Issue 2 would violate CA? If you are referring to calling the `toggleAudioPresenter` from inside the asynchronous thread, then isn't that already happening now? this snippet is present in `checkToggleStateThread`:
``` java
if (!audio.isPlaying()) {
     toggleAudioPresenter.showPlayButton();
}
```
The following changes fix the issue for me, let me know if this doesn't adhere to CA. 
```suggestion
    private void startTimerForShowingPlayButton(PlayableAudio audio) {
        cancelTimerForShowingPlayButton();

        checkToggleStateThread = new Thread(() -> {
            try {
                int timeBetweenChecks = 20;

                while (true) {
                    Thread.sleep(timeBetweenChecks);
                    if (!audio.isPlaying()) {
                        toggleAudioPresenter.showPlayButton();
                    }
                    if (Thread.interrupted()) {
                        return;
                    }
                }
            } catch (InterruptedException ignored) {
            }
        });
        checkToggleStateThread.start();
    }
```

Reply:
@NickinAction I solved this with the dependency injection pattern. Basically, I changed the PlayableAudio interface to introduce a `setOnStopCallback` method where you can set a function to be called whenever the audio stops. In the ToggleAudioInteractor I use this to inject the showPlayButton method, which I believe adheres to CA since there is no direct dependency. It also allows us to eliminate the constant checking code.